src/v/coproc/event_listener.cc

```diff
@@ -196,11 +202,40 @@ ss::future<> event_listener::do_start() {
         } else {
             _active_ids.clear();
             _offset = model::offset(0);
+            co_await advertise_state_update();
         }
     }
+
     co_await do_ingest();
 }
 
+ss::future<> event_listener::boostrap_status_topic() {
+    /// Bootstrap the coproc status topic. At mininum the wasm engine must
+    /// be up for this part of code to be reached.
+    kafka::creatable_topic copro_status_topic{
+      .name = model::coprocessor_status_topic,
+      .num_partitions = 1,
+      .replication_factor = 3,
```

Comment:
should we have more than one partition? will hard coding replication factor of 3 affect single node deployments?

Reply:
Yeah this crossed my mind too. Should we do something more dynamic? Like make a decision on this value after detecting the number of nodes, or something like that.

Reply:
1 is probably ok for now, but maybe making it 2 or 3 is a good trade-off. its not a lot, but it also helps make sure we don't add things that make too many assumptions about the topic configuration. making it a config option would probably be easiest and we can then customize for new deployments. we can do something custom if we need to scale it out later for an existing cluster

Reply:
Ok for now, making this 1